### PR TITLE
Docs: add RegistrationForm model docs

### PR DESCRIPTION
**Pull request by pferreir**
It only took us many years to get there, but I'm finally documenting `RegistrationForm` and friends. It's always very hard to wrap your head about this part and I think this will be a favour to future generations of developers. Also adding mermaid support to the tech docs, which could eventually replace the weird UML stuff we're using.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
 python:
-  version: 3.9
-  setup_py_install: false
-  pip_install: true
+  install:
+    - requirements: docs/requirements.txt
+    - method: pip
+      path: .
+      extra_requirements:
+        - dev
```
Comment (ThiefMaster) [inline .readthedocs.yml:21]: Is that needed? I don't think we have any extras defined there, and the indico package itself shouldn't be installed (I think)

Comment (ThiefMaster) [reply .readthedocs.yml]: ah hm, i thought we had skipped this, but maybe that was in the past when we still had this ZODB crap which failed to install in the RTD env. so nevermind, I think `[dev]` actually makes sense in that case.

**Comment by ThiefMaster** [comment]
just spotted this in conf.py  btw: `copyright = '2021, Indico Team'` - might be nice to just use `date.today().year` there

**Comment by pferreir** [comment]
> date.today().year

I think I'll send another PR to fix a few other things in the docs (e.g. warning messages)

**Pull request merged**